db/seeds.d/050-taxonomies.rb

```diff
@@ -1,23 +1,27 @@
-# Create an initial organization if specified
-if SETTINGS[:organizations_enabled] && ENV['SEED_ORGANIZATION'] && !Organization.any?
-  Organization.without_auditing do
-    original_user, User.current = User.current, User.anonymous_admin
-    @organization = Organization.where(:name => ENV['SEED_ORGANIZATION']).first_or_create
-    User.current = original_user
-  end
-end
+# some associations shouldn't be set or require special handling.
+skip_associations = [:associated_audits, :audits, :default_users, :hosts,
+                     :location_parameters, :organization_parameters,
+                     :taxable_taxonomies ] + Template.descendants.map {|type| type.to_s.tableize.to_sym}
 
-# Create an initial location if specified
-if SETTINGS[:locations_enabled] && ENV['SEED_LOCATION'] && !Location.any?
-  Location.without_auditing do
-    original_user, User.current = User.current, User.anonymous_admin
-    @location = Location.where(:name => ENV['SEED_LOCATION']).first_or_create!
-    User.current = original_user
-  end
-end
+User.as_anonymous_admin do
+  [Location, Organization].select(&:none?).each do |taxonomy|
+    taxonomy.without_auditing do
+      tax_name = ENV.fetch("SEED_#{taxonomy.to_s.upcase}", "Default #{taxonomy}")
+      tax = taxonomy.create!(name: tax_name)
+      associations = taxonomy.reflect_on_all_associations.reject {|assoc| skip_associations.include?(assoc.name)}
+      associations.each do |association|
+        tax.send("#{association.name}=", association.klass.all)
+      end
+
+      # Only default templates are assigned during taxonomy creation
+      Template.where(default: false).each do |template|
```

Comment:
should we associate all templates? if we didn't have any taxonomy before, it behaved like it was available globally right? which is the new default taxonomy purpose

Reply:
The default ones are already assigned when the taxonomy is created by https://github.com/theforeman/foreman/blob/develop/app/models/taxonomy.rb#L12